ibsim/sim_mad.c

```diff
@@ -540,7 +541,7 @@ do_portinfo(Port * port, unsigned op, uint32_t portnum, uint8_t * data)
 			return ERR_BAD_PARAM;	/* trying to change the state of DOWN port */
 
 		if (p->state == 4) {
-			if (p->lid > 0 && p->lid < maxlinearcap
+			if (node->type != SWITCH_NODE && p->lid > 0 && p->lid < maxlinearcap
```

Comment:
Why is node type != switch needed here ?

Reply:
If this is needed, it should be in a separate patch with more info as to why it's needed/how to reproduce.

I'm going to commit this PR without this change.